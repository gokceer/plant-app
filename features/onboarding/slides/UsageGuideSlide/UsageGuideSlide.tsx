import { LinearGradient } from "expo-linear-gradient";
import React from "react";
import { Dimensions, Image, StyleSheet, Text, View } from "react-native";
import { SafeAreaView } from "react-native-safe-area-context";

const { width: screenWidth, height: screenHeight } = Dimensions.get("window");

const UsageGuideSlide = () => {
  return (
    <SafeAreaView style={{ backgroundColor: "white" }}>
      <Text style={styles.title}>
        Get plant <Text style={styles.bold}>care guides</Text>
      </Text>
      <View style={styles.imageWrapper}>
        <Image
          source={require("../../../../assets/images/blurredLeaves.png")}
          style={styles.blurredImage}
          accessibilityLabel="Blurred leaves background"
        />
        <Image
          source={require("../../../../assets/images/decorativeArtwork.png")}
          style={styles.topImage}
          accessibilityLabel="Decorative artwork"
        />
        <Image
          source={require("../../../../assets/images/iPhoneScreen.png")}
          style={styles.image}
          accessibilityLabel="Phone showing care guide"
        />

        <LinearGradient
          colors={["rgba(255,255,255,0)", "rgba(255,255,255,1)"]}
          style={styles.gradientOverlay}
          pointerEvents="none"
        />
      </View>
    </SafeAreaView>
  );
};

export default UsageGuideSlide;

const styles = StyleSheet.create({
  title: {
    marginTop: 12,
    marginBottom: 60,
    fontSize: 28,
    marginLeft: 24,
    fontFamily: "RubikMedium",
  },
  bold: {
    fontFamily: "RubikBold",
  },
  imageWrapper: {
    position: "relative",
    alignItems: "center",
    justifyContent: "center",
  },
  image: {
    width: screenWidth,
    height: screenHeight * 0.75,
    resizeMode: "contain",
  },
  topImage: {
    position: "absolute",
    width: screenWidth,
    height: screenHeight * 0.4,
    top: -50,
    zIndex: 1,
  },
  blurredImage: {
    position: "absolute",
    width: screenWidth,
    height: screenHeight * 0.4,
    top: -15,
    zIndex: -1,
  },
  gradientOverlay: {
    position: "absolute",
    bottom: 0,
    left: 0,
    right: 0,
    width: "100%",
    height: 120,
  },
});
