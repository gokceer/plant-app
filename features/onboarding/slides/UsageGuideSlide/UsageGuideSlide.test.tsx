import { render } from "@testing-library/react-native";
import React from "react";
import UsageGuideSlide from "./UsageGuideSlide";

jest.mock("expo-linear-gradient", () => {
  const { View } = require("react-native");
  return {
    LinearGradient: View,
  };
});

describe("UsageGuideSlide", () => {
  it("renders correctly and matches snapshot", () => {
    const { toJSON } = render(<UsageGuideSlide />);
    expect(toJSON()).toMatchSnapshot();
  });
});
