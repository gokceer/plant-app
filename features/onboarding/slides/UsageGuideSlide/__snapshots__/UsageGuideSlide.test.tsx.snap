// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsageGuideSlide renders correctly and matches snapshot 1`] = `
<RNCSafeAreaView
  edges={
    {
      "bottom": "additive",
      "left": "additive",
      "right": "additive",
      "top": "additive",
    }
  }
  style={
    {
      "backgroundColor": "white",
    }
  }
>
  <Text
    style={
      {
        "fontFamily": "RubikMedium",
        "fontSize": 28,
        "marginBottom": 60,
        "marginLeft": 24,
        "marginTop": 12,
      }
    }
  >
    Get plant 
    <Text
      style={
        {
          "fontFamily": "RubikBold",
        }
      }
    >
      care guides
    </Text>
  </Text>
  <View
    style={
      {
        "alignItems": "center",
        "justifyContent": "center",
        "position": "relative",
      }
    }
  >
    <Image
      accessibilityLabel="Blurred leaves background"
      source={
        {
          "testUri": "../../../assets/images/blurredLeaves.png",
        }
      }
      style={
        {
          "height": 533.6,
          "position": "absolute",
          "top": -15,
          "width": 750,
          "zIndex": -1,
        }
      }
    />
    <Image
      accessibilityLabel="Decorative artwork"
      source={
        {
          "testUri": "../../../assets/images/decorativeArtwork.png",
        }
      }
      style={
        {
          "height": 533.6,
          "position": "absolute",
          "top": -50,
          "width": 750,
          "zIndex": 1,
        }
      }
    />
    <Image
      accessibilityLabel="Phone showing care guide"
      source={
        {
          "testUri": "../../../assets/images/iPhoneScreen.png",
        }
      }
      style={
        {
          "height": 1000.5,
          "resizeMode": "contain",
          "width": 750,
        }
      }
    />
    <View
      colors={
        [
          "rgba(255,255,255,0)",
          "rgba(255,255,255,1)",
        ]
      }
      pointerEvents="none"
      style={
        {
          "bottom": 0,
          "height": 120,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "width": "100%",
        }
      }
    />
  </View>
</RNCSafeAreaView>
`;
